src/js/util/object.js

```diff
@@ -0,0 +1,53 @@
+function isObject(something) {
+  return typeof something === 'object';
+}
+
+/**
+ *
+ * @param {Object} object
+ */
+function cloneDeepObject(object) {
+  const keys = Object.keys(object);
+  return Object.freeze(
+    keys.reduce((result, key) => {
+      const value = object[key];
+      if (Array.isArray(value)) {
+        return { ...result, [key]: [...value] };
+      }
+      if (typeof value === 'object') {
+        return { ...result, [key]: Object.freeze({ ...cloneDeepObject(value) }) };
+      }
+      return { ...result, [key]: value };
+    }, {})
+  );
+}
+
+/**
+ *
+ * @param {[]} array
+ */
+function cloneDeepArray(array) {
+  return array.reduce((result, item) => {
+    if (Array.isArray(item)) {
+      return [...result, ...cloneDeepArray(item)];
+    }
+    if (typeof value === 'object') {
+      return [...result, cloneDeepObject(item)];
+    }
+    return [...result, item];
+  }, []);
+}
```

Comment:
```suggestion
function cloneDeepObject(object) {
  return Object.entries(object)
            .reduce((result, [key, value]) => ({
              ...result,
              [key]: cloneDeep(value)
            }), {});
}

function cloneDeepArray(array) {
  return array.map(cloneDeep);
}
```
생각해보니, 이렇게 표현해도 되겠네요 ㅎㅎ

Reply:
해당 부분은 좀 더 바람직한 방법을 찾았습니다.
(이전에 제가 제시해 드렸던 방향이 아니라 리뷰어님께서 제시해 주신 방향으로 수정해볼 수 있게 되었습니다)
객체 깊은 복사 -> 객체 깊은 Freezing -> 반환

상수를 freezing해야 하기 때문에 freezing하는 과정은 별개로 분리하는 것이 타당하다는 판단이 들었습니다.
보완해서 리뷰 요청드릴게요!

Reply:
위에랑 중복된 피드백이였네요 ㅠㅠ